.github/workflows/e2e-connectivity.yaml

```
@@ -0,0 +1,222 @@
+name: NetworkPolicy tests
+
+on:
+  push:
+    branches: [ "main" ]
+
+  pull_request:
+    branches: [ "main", "release-*" ]
+
+defaults:
+  run:
+    shell: bash
+    working-directory: ./src/knative.dev/serving
+
+jobs:
+  e2e-connectivity:
+    name: e2e connectivity
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s-version:
+        - v1.22.0
+
+        test-suite:
+        - ./test/e2e
+
+        include:
+        - k8s-version: v1.22.0
+          kind-version: v0.11.1
+          kingress: istio
+          kind-image-sha: sha256:f97edf7f7ed53c57762b24f90a34fad101386c5bd4d93baeb45449557148c717
+          cluster-suffix: c${{ github.run_id }}.local
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: on
+      GOFLAGS: -tags=nostackdriver
+      # https://github.com/google/go-containerregistry/pull/125 allows insecure registry for
+      # '*.local' hostnames. This works both for `ko` and our own tag-to-digest resolution logic,
+      # thus allowing us to test without bypassing tag-to-digest resolution.
+      REGISTRY_NAME: registry.local
+      REGISTRY_PORT: 5000
+      KO_DOCKER_REPO: registry.local:5000/knative
+    steps:
+    - name: Set up Go 1.16.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16.x
```

1.16 is going to be deprecated soon - let's bump to 1.17

```suggestion
    steps:
    - name: Set up Go 1.17.x
      uses: actions/setup-go@v2
      with:
        go-version: 1.17.x
```